onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -153,6 +154,20 @@ Status GraphPartitioner::Partition(Graph& graph, bool export_dll, FuncManager& f
     std::vector<Node*> nodes_need_compile;
     std::vector<std::unique_ptr<ComputeCapability>> capabilities =
         provider->GetCapability(graph_viewer, kernel_registry_mgr_.GetKernelRegistriesByProviderType(provider->Type()));
+
+    // Excludce the subgraph that is prefered to be place in CPU
+    // These are usually shape related computation subgraphs
```

Comment:
Compiler based EPs usually handles shape computation nodes internally as no-op if the compiled code inlined the shape info. Splitting those ops out for CPU is not preferred.

Reply:
As suggested by George, limiting the effect to CUDA, TensorRT and MIGraphX EPs.

Reply:
actually, let's keep it with CUDA EP for now. We should examine those other 2 more carefully.
(will point amd migraphx dev to it as well)
As ke mentions, it might be optimized by the EP.

Reply:
OK. I will keep it CUDA only...  but leave it in GraphPartitioner.

Reply:
Alternatively, we could have these code in a shared util function for all EPs, instead of baking them in GraphPartitioner. Given that IExecutionProvider::GetCapability() is intended to let EP decides node ownership, it would be consistent in design if we let EP invoke utils for CPU placement in its GetCapability().

Reply:
As suggested, I have made this an util function that can be invoked from GetCapability()